### ISSUE TITLE
[no-unsafe-negation] Duplicates TypeScript's TS2358, TS2360, and TS2365 errors (?)

### ISSUE THREAD

**Issue by glen-84** [typescript-eslint/typescript-eslint#1146]
**Repro**

```JSON
{
  "rules": {
    "no-unsafe-negation": ["error", { "enforceForOrderingRelations": true }],
  }
}
```

```TS
const a = 1;
const b = 2;
let foo;

if (!a < b) {
    /**/
}

while (!a > b) {
    /**/
}

foo = !a <= b;

foo = !a >= b;

const myObject = {};
const key = "";
const Ctor = Function;

if (!key in myObject) {
    // operator precedence makes it equivalent to (!key) in object
    // and type conversion makes it equivalent to (key ? "false" : "true") in object
}

if (!myObject instanceof Ctor) {
    // operator precedence makes it equivalent to (!obj) instanceof Ctor
    // and it equivalent to always false since boolean values are not objects.
}
```

**Expected Result**

TS errors only.

**Actual Result**

TS and ESLint errors.

![image](https://user-images.githubusercontent.com/261509/67637469-a752d900-f8e3-11e9-8099-7510c9bda423.png)

**Additional Info**

`enforceForOrderingRelations` is actually off by default, so technically `TS2365` is probably not being duplicated.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.5.0` |
| `@typescript-eslint/parser`        | `2.5.0` |
| `TypeScript`                       | `3.5.3` |
| `ESLint`                           | `6.6.0` |
| `node`                             | `12.13.0` |
| `npm`                              | `6.12.0` |

**Comment by bradzacher** [comment]
Sorry, what is the request here?
You don't want the rule to report errors that TS reports? 

I don't think it's a problem reporting the same errors that TS does, because a lot of people don't use the TS compiler in their build (eg babel).

**Comment by glen-84** [comment]
I thought that this was the same type of issue as #983 and #999?

**Comment by bradzacher** [comment]
Ah I'm sorry, I didn't realise you were referring to the recommended set.
Yup makes sense.

**Comment by bradzacher** [comment]
consolidated eslint-recommended changes in #1273
